product_print_category/tests/test_product_print_category.py

```diff
@@ -136,3 +136,60 @@ def test_21_onchange(self):
         product.categ_id = self.env.ref("product.product_category_all")
         product._onchange_categ_id_company_id()
         self.assertEqual(product.print_category_id.id, False)
+
+    def test_30_test_wizard_on_change_products_template(self):
+        # Add a new product template
+        template = self.ProductTemplate.create(
+            {
+                "name": "Demo Product Template Name",
+                "print_category_id": self.print_category_1.id,
+            }
+        )
+
+        # Set all products as printed
+        self.print_category_1.mapped("product_to_print_ids").write({"to_print": False})
+        self.assertEqual(template.to_print, False)
+
+        # Change name of one product template only
+        template.name = "Demo Product Template New Name"
+        products = self.ProductProduct.search(
+            [
+                ("print_category_id", "=", self.print_category_1.id),
+                ("to_print", "=", True),
+            ]
+        )
+        self.assertEqual(len(products), 1)
+        self.assertTrue(template.to_print)
+
+    def test_31_test_wizard_on_change_products_products(self):
+        # Add a new product
+        product = self.ProductProduct.create(
+            {
+                "name": "Demo Product Name",
+                "print_category_id": self.print_category_1.id,
+            }
+        )
+        second_product = self.ProductProduct.create(
+            {
+                "name": "Demo Second Product Name",
+                "print_category_id": self.print_category_1.id,
+                "product_tmpl_id": product.product_tmpl_id.id,
+            }
+        )
+
+        # Set all products as printed
+        self.print_category_1.mapped("product_to_print_ids").write({"to_print": False})
+        self.assertEqual(product.to_print, False)
+        self.assertEqual(second_product.to_print, False)
+
+        # Change name of product, shared with second_product
+        product.name = "Demo Product New Name"
```

Comment:
Hi. thanks for all the changes. just a question. why not better used some field NOT shared between many products of the same template. 
if so, we can verify that if we change the information of a variant, it doesn't change the information of the other variants of the same template.

i think you can use variant price field.